### PR TITLE
fix: can not read .swcrc when transformOpts is empty object

### PR DESCRIPTION
**Pull request by IronLu233**

### CODE REVIEW

**Comment by IronLu233** [comment]
Should not change `tsx` params in @swc/jest, the better way is change it in @swc/core, because this feature has implementd but not use for some methods

**Comment by IronLu233** [comment]
@kdy1 Please review it

File: package.json [hunk 1]
```diff
@@ -16,11 +16,11 @@
     "access": "public"
   },
   "peerDependencies": {
-    "@swc/core": "^1.2.30"
+    "@swc/core": "*"
   },
   "devDependencies": {
-    "@swc/core": "^1.2.77",
-    "@types/node": "^14.11.1",
+    "@swc/core": "*",
```
Comment (kdy1) [inline package.json:22]: I'm not sure about this. Why is `*` used?

File: index.ts [hunk 2]
```diff
@@ -30,25 +30,37 @@ function getJestTransformConfig(
...
         set(swcOptions, 'jsc.transform.hidden.jest', true)
 
         transformOpts = swcOptions
       }
 
+
+      const parserConfig = transformOpts.jsc?.parser;
+
+      // typescript: `tsx` will be modified if file extension is `ts`.
+      if (parserConfig?.syntax === 'typescript') {
```
Comment (kdy1) [inline index.ts:60]: As we pass filename, we can blindly set `tsx: true`. Rust code will disable `tsx` if extension is `.ts`

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: index.ts (hunk 2 above)
Comment (IronLu233) [reply index.ts:60]: I found this feature, but seems it not work when we call `transformSync`. Seems fix it in rust side is better.

File: package.json (hunk 1 above)
Comment (IronLu233) [reply package.json:22]: I thought the @swc/core should keep the latest version, but here, manually upgrade it is better~. I will revert this change

**Pull request merged**